src/scores/probability/brier_impl.py

```diff
@@ -61,3 +67,155 @@ def brier_score(
+    obs: XarrayLike,  # type: ignore
+    ensemble_member_dim: str,
+    thresholds: Union[float, int, Sequence[float]],
+    *,  # Force keywords arguments to be keyword-only
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+    weights: Optional[xr.DataArray] = None,
+    fair_correction: bool = True,
+    threshold_mode: Literal[">=", ">"] = ">=",
+) -> XarrayLike:  # type: ignore
+    """
+    Calculates the Brier score for an ensemble forecast for the provided thresholds. By default,
+    the fair Brier score is calculated, which is a modified version of the Brier score that
+    applies a correction related to the number of ensemble members and the number of
+    ensemble members that predict the event.
+
+    The fair Brier score for ensembles is defined for a single event threshold as:
+
+
+    .. math::
+        s_{i,y} = \\left(\\frac{i}{m} - y\\right)^2 - \\frac{i(m-i)}{m^2(m-1)}
+
+    Where:
+
+    - :math:`i` is the number of ensemble members that predict the event
+    - :math:`m` is the total number of ensemble members
+    - :math:`y` is the observed event
+
+    When the fair correction is not applied (i.e., ``fair_correction=False``),
+    the Brier score is calculated as:
+
+
+    .. math::
+        s_{i,y} = \\left(\\frac{i}{m} - y\\right)^2
+
+    If ``fair_correction`` is set to ``True`` and the number of ensemble members is 1, the
+    fair correction will not be applied to avoid division by zero. If you want to
+    set the minimum number ensemble members required to calculate the Brier score,
+    we recommend preprocessing your data to remove data points with fewer than the
+    minimum number of ensemble members that you want.
+
+    Args:
+        fcst: Forecast or predicted variables in xarray.
+        obs: Observed variables in xarray.
+        ensemble_member_dim: The dimension name of the ensemble member.
+        thresholds: The threshold(s) to use for the Brier score that define what is
+            considered an event. If multiple thresholds are provided, they should
+            be monotonically increasing with no NaNs
+        reduce_dims: Optionally specify which dimensions to reduce when
+            calculating the Brier score. All other dimensions will be preserved.
+        preserve_dims: Optionally specify which dimensions to preserve when
+            calculating the Fair Brier score. All other dimensions will be reduced. As a
+            special case, 'all' will allow all dimensions to be preserved. In
+            this case, the result will be in the same shape/dimensionality
+            as the forecast, and the errors will be the absolute error at each
+            point (i.e. single-value comparison against observed), and the
+            forecast and observed dimensions must match precisely.
+        weights: Optionally provide an array for weighted averaging (e.g. by area, by latitude,
```

Comment:
Can you specify that it is taking a weighted average of the scores for each forecast case? You could be taking weighted averages of other quantities